File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p04039/p04039_num6_parsed.cpp
#include<iostream>
#include<vector>

using namespace std;

bool containsDislikedDigit(int amount, vector<int>& dislikedDigits) {
    while(amount > 0) {
        int digit = amount % 10;
        if(find(dislikedDigits.begin(), dislikedDigits.end(), digit) != dislikedDigits.end()) {
            return true;
        }
        amount /= 10;
    }
    return false;
}

int main() {
    int N, K;
    cin >> N >> K;

    vector<int> dislikedDigits(K);
    for(int i = 0; i < K; i++) {
        cin >> dislikedDigits[i];
    }
    
    int amount = N;
    while(containsDislikedDigit(amount, dislikedDigits)) {
        amount++;
    }
    
    cout << amount << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p04039/p04039_num6_parsed.py
N, K = map(int, input().split())
disliked_digits = set(map(int, input().split()))

def has_disliked_digits(num):
    while num > 0:
        digit = num % 10
        if digit in disliked_digits:
            return True
        num //= 10
    return False

amount = N
while True:
    if not has_disliked_digits(amount):
        break
    amount += 1

print(amount)
